src/main/java/com/ekart/service/ProductService.java

```diff
@@ -44,7 +57,55 @@ public Product getProductById(Long id) throws RecordNotFoundException {
         }
     }
 
-    public Product saveProduct(Product product) {
+    public Product saveProduct(Map<String, Object> request) {
```

Comment:
rename request to something else..

In general, request and response keywords should not be used in the service file. In controller it is fine.

Reply:
done